syntax = "proto3";
package agent;




import "agent_admin_sdk/model/agent_admin/plugin_instance.proto";

/*
ListAgentPlugins请求
*/
message ListAgentPluginsRequest {
    /*
    页码
    */
    int32 page = 1 ;
    /*
    每页数据量
    */
    int32 pageSize = 2 ;
    /*
    实例唯一标识
    */
    string id = 3 ;
}

/*
ListAgentPlugins返回
*/
message ListAgentPluginsResponse {
    /*
    页数
    */
    int32 page = 1 ;
    /*
    页大小
    */
    int32 page_size = 2 ;
    /*
    总数
    */
    int32 total = 3 ;
    /*
    数据列表
    */
    repeated agent_admin.PluginInstance list = 4 ;
}

/*
ListAgentPluginsApi返回
*/
message ListAgentPluginsResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    agent.ListAgentPluginsResponse data = 4 ;
}

